Sources/General/Commands/Setup.swift

```diff
@@ -0,0 +1,258 @@
+//
+//  Copyright © 2020 Rosberry. All rights reserved.
+//
+
+import Foundation
+import ArgumentParser
+import ZIPFoundation
+
+final class Setup: ParsableCommand {
+
+    enum Error: Swift.Error {
+        case githubName(_ github: String)
+        case url(_ url: String)
+        case download(_ url: URL)
+        case write(_ url: URL)
+        case remove(_ url: URL)
+    }
+
+    static let configuration: CommandConfiguration = .init(commandName: "setup", abstract: "Provides your environment with templates")
+
+    @Option(name: .shortAndLong,
+            default: Constants.defaultTemplatesGithub,
+            help: .init(stringLiteral:
+                "Use this option if templates are placed on github." +
+                " Format: \"<github>\\ [branch]\". Default: \"\(Constants.defaultTemplatesGithub)\""))
+    var github: String
```

Comment:
I don't think that this is a good name for this variable. It isn't describe what it is at all. Maybe the better way is name it more detailed and specify `@Option(name: [.short("g"), .long("github")])`?